backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendVmPoolsResource.java

```diff
@@ -118,6 +118,9 @@ public Response add(VmPool pool) {
         AddVmPoolParameters params = new AddVmPoolParameters(entity, vm, size);
         params.setConsoleEnabled(pool.isSetVm() && pool.getVm().isSetConsole() && pool.getVm().getConsole().isSetEnabled() ?
                 pool.getVm().getConsole().isEnabled() : !getConsoleDevicesForEntity(template.getId()).isEmpty());
+        if (pool.isSetVm()) {
```

Comment:
what will happen if this isn't true and we won't enter here?

Reply:
Graphic types from the template will be used.

Reply:
it would also be nice if we can move other param setter into this if block and drop the check on the setter, like `params.setConsoleEnabled(pool.isSetVm() &&` above and `params.setRngDevice(pool.isSetVm() &&` below.

Reply:
I've tried to do that and this doesn't make the code more concise or readable. I can move `pool.isSetVm()` condition out, but there is the second part of the condition that is still there. So it is still a conditional expression and I also need to handle the case, when `pool.isSetVm()` is false separately. Or I can move setting the parameters to their default values before the `if` statement, but this means additional useless calculations (and some of them include a query).